retrieval/discovery/consul.go

```diff
@@ -95,6 +96,17 @@ func NewConsulDiscovery(conf *config.ConsulSDConfig) *ConsulDiscovery {
 		scrapedServices: map[string]struct{}{},
 		services:        map[string]*consulService{},
 	}
+	// If the datacenter isn't set in the clientConf, let's get it from the local Consul agent
+	// (Consul default is to use local node's datacenter if one isn't given for a query)
```

Comment:
Yeah, this and some other things would be great to have in `make` or some other stage of the verification pipeline at some point :)